lib/setup/setupInstall.js

```diff
@@ -754,34 +754,38 @@ function Install(options) {
             }
         }
 
-        // install linux/osx libraries
-        packetManager = packetManager || new PacketManager();
-        packetManager.install(adapterConf.common.osDependencies && adapterConf.common.osDependencies[process.platform])
-            .catch(e => console.error(e.message))
-            .then(() => {
-                if (!fs.existsSync(adapterDir + '/node_modules')) {
-                    // Install node modules
-                    installNpm(adapter, (err, _adapter) => {
-                        if (err) {
-                            processExit(err);
-                        } else {
-                            upload.uploadAdapter(_adapter, true, true, null, null, () =>
-                                upload.uploadAdapter(_adapter, false, true, null, null,() =>
-                                    callInstallOfAdapter(_adapter, adapterConf, () =>
-                                        that.uploadStaticObjects(adapter, _err =>
-                                            upload.upgradeAdapterObjects(adapter, () =>
-                                                callback(adapter))))));
-                        }
-                    });
+        if (adapterConf.common.osDependencies && adapterConf.common.osDependencies[process.platform]) {
+            // install linux/osx libraries
+            try {
+                packetManager = packetManager || new PacketManager();
```

Comment:
if no packetmanager detected, this could throw because of https://github.com/ioBroker/ioBroker.js-controller/pull/1223/files#diff-35c66992201af88561434473c75bd5ad836a582d6b3611fa4f400e4d5627ac6bL84